### PR TITLE
feat(api): rds support for hasmany, hasone and belongsto directives

### PR DESCRIPTION
**Pull request by sundersc**
#### Description of changes

Adds support for hasMany, hasOne and belongsTo directives for RDS datasource.

- Adds new attribute references to mention fields used for JOIN.
- Uses existing layer LIST operation to fetch the child records.
- Update layer version to the latest.
- Fix a bug with vpc config.

(Example schema from E2E test)
```graphql
type Blog @model {
  id: String! @primaryKey
  content: String
  posts: [Post] @hasMany(references: ["blogId"])
}
type Post @model {
  id: String! @primaryKey
  content: String
  blogId: String!
  blog: Blog @belongsTo(references: ["blogId"])
}
type User @model {
  id: String! @primaryKey
  name: String
  profile: Profile @hasOne(references: ["userId"])
}
type Profile @model {
  id: String! @primaryKey
  details: String
  userId: String!
  user: User @belongsTo(references: ["userId"])
}
```
##### CDK / CloudFormation Parameters Changed

NA

#### Issue #, if available

NA

#### Description of how you validated changes
- E2E Test
- Manual Test
- Unit tests

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] Any CDK or CloudFormation parameter changes are called out explicitly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-relational-transformer/src/resolver/rds-generator.ts [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+  toJson,
+  iff,
+  not,
+} from 'graphql-mapping-template';
+import { BelongsToDirectiveConfiguration, HasManyDirectiveConfiguration, HasOneDirectiveConfiguration } from '../types';
+import { RelationalResolverGenerator } from './generator';
+
+const CONNECTION_STACK = 'ConnectionStack';
+
+export class RDSRelationalResolverGenerator extends RelationalResolverGenerator {
```
Comment (sundersc) [inline packages/amplify-graphql-relational-transformer/src/resolver/rds-generator.ts:25]: Most of the code on this file moved from `resolver.ts` to DDB specific resolver generator class.

File: packages/amplify-graphql-relational-transformer/src/graphql-belongs-to-transformer.ts [hunk 2]
```diff
@@ -185,7 +208,7 @@ const validate = (config: BelongsToDirectiveConfiguration, ctx: TransformerConte
     });
   });
 
-  if (!isBiRelation) {
+  if (!isBiRelation && dbType === DDB_DB_TYPE) {
     throw new InvalidDirectiveError(
       `${config.relatedType.name.value} must have a relationship with ${object.name.value} in order to use @${directiveName}.`,
```
Comment (phani-srikar) [inline packages/amplify-graphql-relational-transformer/src/graphql-belongs-to-transformer.ts:213]: wouldn't we have to check this for RDS DB types as well?

File: packages/amplify-graphql-relational-transformer/src/schema.ts [hunk 3]
```diff
@@ -127,10 +127,14 @@ const ensureModelSortDirectionEnum = (ctx: TransformerContextProvider): void =>
...
-  const { field, fieldNodes, object, relatedType } = config;
+  const { field, fieldNodes, references, object, relatedType } = config;
 
   // If fields were explicitly provided to the directive, there is nothing else to do here.
-  if (fieldNodes.length > 0) {
+  if (fieldNodes?.length > 0) {
+    return;
+  }
+
+  if (references && references.length > 0) {
```
Comment (phani-srikar) [inline packages/amplify-graphql-relational-transformer/src/schema.ts:137]: nit: `fieldNodes?.length > 0 || references?.length > 0` in one shot.

**Review by phani-srikar — APPROVED** [review]

File: packages/amplify-graphql-relational-transformer/src/graphql-belongs-to-transformer.ts (hunk 2 above)
Comment (sundersc) [reply packages/amplify-graphql-relational-transformer/src/graphql-belongs-to-transformer.ts:213]: Not required. RDS doesn't require bi directional relationship. Single field defines the relationship on both ends, so customer can use belongsTo without hasMany or hasOne.

**Review by alharris-at — APPROVED** [review]

**Pull request merged**